### PR TITLE
Strip out carriage returns

### PR DESCRIPTION
**Pull request by teoljungberg**
References #1008

### CODE REVIEW

File: plugin/fugitive.vim [hunk 1]
```diff
@@ -2511,6 +2511,7 @@ function! s:ReplaceCmd(cmd,...) abort
   silent exe 'keepalt file '.tmp
   try
     silent edit!
+    silent exec '%g/'.nr2char(13).'/d'
```
Comment (teoljungberg) [inline plugin/fugitive.vim]: This is a very naive approach, I'd gladly take pointers!

Comment (tpope) [reply plugin/fugitive.vim]: First pointer is don't delete the entire line.

Comment (teoljungberg) [reply plugin/fugitive.vim]: 🤕 

Fixed in 04ee399 with a normal substitution.

Comment (teoljungberg) [reply plugin/fugitive.vim]: I also updated it to use the `c` s_flag to only remove the carriage returns if they exist.

**Comment by teoljungberg** [comment]
Any thoughts on this @tpope? I've been running this branch locally since it was opened without any issues, if you can find any edge-cases for this implementation - or a better implementation - I'm all ears.

File: plugin/fugitive.vim (hunk 1 above)
Comment (teoljungberg) [reply plugin/fugitive.vim]: Edit, not `c` but `e`

**Comment by tpope** [comment]
Thinking something like `1,/^diff --git\|\%$/g/\r$/s///`, to avoid munging the diff.

**Comment by teoljungberg** [comment]
TIL that you can search for a patterns for ex-commands. Thanks!

I rebased and squashed everything into two commits, one for the unused variable - the other for stripping out the carriage returns.

**Comment by tpope** [comment]
Just noticed you're doing this to all sorts of buffers. Try it before https://github.com/tpope/vim-fugitive/blob/3bdac8b60383c0326991ba08bfd57d4e70925f3a/plugin/fugitive.vim#L2732. Based on the context you'll want to add a `keepjumps`.

**Pull request merged**